### PR TITLE
[RHCLOUD-19374] Migration add table users

### PR DESCRIPTION
**Pull request by petracihalova**
For Single User based Sources we need to add new table `users`:
- columns: id, user_id, tenant_id, created_at, updated_at
- primary key: id
- foreign key: tenant_id (table `tenants`)
- all fields should be not null
- user_id should be unique
- combination user_id + tenant_id should be unique

Then we need new model `User`

**JIRA:** [RHCLOUD-19374](https://issues.redhat.com/browse/RHCLOUD-19374)

### CODE REVIEW

File: db/migrations/20220603140800_add_table_users.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package migrations
+
+import (
+	_ "embed"
```
Comment (lpichler) [inline db/migrations/20220603140800_add_table_users.go]: I would say this is not necessary

File: db/migrations/20220603140800_add_table_users.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	"gorm.io/gorm"
+)
+
+func AddTableUsers() *gormigrate.Migration {
+	type Tenant struct {
+		Id int64
+	}
+
+	type User struct {
+		Id     int64  `gorm:"primarykey"`
```
Comment (MikelAlejoBR) [inline db/migrations/20220603140800_add_table_users.go:17]: I think this needs the `autoIncrement` tag too, right?

**Review by MikelAlejoBR — APPROVED** [review]
LGTM!

File: db/migrations/20220603140800_add_table_users.go (hunk 1 above)
Comment (petracihalova) [reply db/migrations/20220603140800_add_table_users.go]: tnx, done

File: db/migrations/20220603140800_add_table_users.go (hunk 2 above)
Comment (petracihalova) [reply db/migrations/20220603140800_add_table_users.go:17]: gorm do this automatically for numeric primary keys

```
sources_api_development=# \d public.users
                                            Table "public.users"
   Column   |            Type             | Collation | Nullable |                 Default
------------+-----------------------------+-----------+----------+------------------------------------------
 id         | bigint                      |           | not null | nextval('public.users_id_seq'::regclass)
 user_id    | text                        |           | not null |
 tenant_id  | bigint                      |           | not null |
 created_at | timestamp without time zone |           | not null |
 updated_at | timestamp without time zone |           | not null |
Indexes:
    "users_pkey" PRIMARY KEY, btree (id)
    "users_user_id_idx" UNIQUE, btree (user_id)
    "users_user_id_tenant_id_idx" UNIQUE, btree (user_id, tenant_id)
Foreign-key constraints:
    "fk_users_tenant" FOREIGN KEY (tenant_id) REFERENCES public.tenants(id)
```

**Comment by petracihalova** [comment]
comments solved, I added to `created_at` and `updated_at` fields the property "_timestamp without timezone_" to have it same as in other tables

```
sources_api_development=# \d public.users
                                            Table "public.users"
   Column   |            Type             | Collation | Nullable |                 Default
------------+-----------------------------+-----------+----------+------------------------------------------
 id         | bigint                      |           | not null | nextval('public.users_id_seq'::regclass)
 user_id    | text                        |           | not null |
 tenant_id  | bigint                      |           | not null |
 created_at | timestamp without time zone |           | not null |
 updated_at | timestamp without time zone |           | not null |
Indexes:
    "users_pkey" PRIMARY KEY, btree (id)
    "users_user_id_idx" UNIQUE, btree (user_id)
    "users_user_id_tenant_id_idx" UNIQUE, btree (user_id, tenant_id)
Foreign-key constraints:
    "fk_users_tenant" FOREIGN KEY (tenant_id) REFERENCES public.tenants(id)
```

File: db/migrations/20220603140800_add_table_users.go (hunk 2 above)
Comment (MikelAlejoBR) [reply db/migrations/20220603140800_add_table_users.go:17]: Aaaah great then!

**Review by lpichler — APPROVED** [review]

**Pull request merged**